store/src/java/com/zimbra/cs/html/owasp/OwaspHtmlSanitizer.java

```diff
@@ -79,6 +81,52 @@ private String checkUnbalancedTags(String html) {
         }
         return html;
     }
+    
+    /** 
+     * A method to check & remove any any child parent comparator inside <style> tag 
+     * 
+     * @return the Formatted Html after removing child comparators(>)
+     */
+    private String removeChildComaparator(String html) {
```

Comment:
what would be side effect of removing this child comparators?

Reply:
The particular css property will not apply I guess . But it is not reflecting much on mime . I have checked with multiple mimes but I am not getting any difference or negligence differencemay come .  

Exp - in one mime of 2744 there is a line ( #portal-page-1### >div>div{padding-top:5px;} )
          I am just replacing this line with ( #portal-page-1div>div{padding-top:5px;} )
          
 So here property will add . Sanitize is happening correctly & for this ('>') one comparator only we are getting null body in mail

Reply:
I'm worried about the side effect. If it is related to java html sanitizer which we are using, https://github.com/Zimbra/java-html-sanitizer-release-20190610.1/ , has we reported the issued to the official one https://github.com/OWASP/java-html-sanitizer and got any response? I don't think it is Zimbra-specific issue...

Reply:
ya @k-kato .. Akshat has reported already regarding this long time back - https://github.com/OWASP/java-html-sanitizer/issues/251 ( ZBUG-2639 )
But as we are not getting proper response he has suggested in his last comment to implement it from our side & it will be a temporary fix until OWASP library or people fix this issue , we have 2 3 tickets also related to this bug( it is coming as blocker for a costumer as email body will come as null )

Reply:
I got it. Thanks.

Reply:
@rupeshtelus I guess this is not the correct place to handle this, if you want to fix the OWASP-specific bugs, then create a PR for the fix in https://github.com/Zimbra/java-html-sanitizer-release-20190610.1/ repo as @k-kato mentioned about its side-effects.

Please do not mix the fixes of OWASP bugs into the Zimbra mailbox repo as a temporary resolution as it is an OpenSource project and can hamper down the line. In the past, we have handled OWASP bug fixes in that particular OWASP repo and each time its version is bumped, please go through the OWASP repo on how to fix an OWASP bug with test cases and integrate it with the Zimbra mailbox repo for more details.